Lib9c/Action/ValidatorDelegation/ReleaseValidatorUnbondings.cs

```diff
@@ -52,5 +64,64 @@ public override IWorld Execute(IActionContext context)
 
             return repository.World;
         }
+
+        private IWorld Unstake(IWorld world, IActionContext context, Address address)
+        {
+            var agentAddress = new AgentAddress(address);
+            var guildRepository = new GuildRepository(world, context);
+            if (guildRepository.TryGetGuildParticipant(agentAddress, out var guildParticipant))
+            {
+                var guild = guildRepository.GetGuild(guildParticipant.GuildAddress);
+                var goldCurrency = world.GetGoldCurrency();
+                var stakeStateAddress = guildParticipant.DelegationPoolAddress;
+                var gg = world.GetBalance(stakeStateAddress, guild.DelegationCurrency);
+                if (gg.Sign > 0)
+                {
+                    var (ncg, _) = ConvertToGoldCurrency(gg, goldCurrency);
+                    world = world.BurnAsset(context, stakeStateAddress, gg);
+                    world = world.TransferAsset(
+                        context, stakeStateAddress, address, ncg);
+                }
+            }
+
+            return world;
+        }
+
+        private static (FungibleAssetValue Gold, FungibleAssetValue Remainder)
+            ConvertToGoldCurrency(FungibleAssetValue fav, Currency targetCurrency)
+        {
```

Comment:
Can you suggest a mathematically simple example code, this is my limit.

Reply:
> Not sure, but I think this can be done easier by just multiplying or dividing by decimal places difference powered by 10.

Amended https://github.com/planetarium/lib9c/pull/2961/commits/45e56e618baff7bc27ae90f5bfef9e1e9ca08960